# Introducción 

El resumen de datos proporciona una forma conveniente para resumir la distribución de una muestra de datos a partir de una determinados valores estadísticos. Estadísticos como la media y la desviación estándar son usados para resumir aquellos datos que tienen una distrubición Gausiana, pero estos podrían no tener sentido o incluso podrían darnos información erronea, si nuestra muestra de datos tiene una distribución Gausiana.

# Resumen muestras de datos no paramétricas

Las técnicas para resumir datos proporcionan una forma de describir la distribución de datos usando algunas medidas estadísticas. El ejemplo más clásico es el resumen de datos de una distribución Gausiana mediante las medidas estadísticas de la media y la desviación estándar. El problema recae cuando nuestros datos no tiene una distribución Gausiana. Técnicamente podemos calcular estos parámetros para dicha muestra, sin embargo, las conclusiones no serán correctas e incluso podrían dar lugar a generar errores. En caso de que nuestra muestra no tenga una distribución normal podemos hacer usos de los 5-números.

# Los 5-números

El resumen de los cinco números, es una técnica de resumen para datos no paramétricos. Puede ser usado para describir las distribuciones de una muestra de datos independientemente de la forma que tenga dicha distribución. Los cinco parámetro estadísticos que componen este resumen son:

* **Mediana** : se trata del valor central de una muestra, es también conocido como el percentil 50th o el 2nd percentil.

* **1st Quartile** : percentil 25th.

* **3rd Quartile** : percentil 75th.

* **Mínimo** : se trata del valor mínimo de una muestra.

* **Máximo** : se trata del valor máximo de una muestra.

Un cuartil es un valor observado que nos permite separar una muestra ordenada en 4 partes iguales. La mediana, a segundo cuartil, divide la muestra ordenada en dos partes iguales, y el primer y tercer cuartil divide cada una de estas medias en cuartos.

Un percentil es un valor observado que nos permite separar la muestra ordenada en 100 partes iguales. Los cuartiles son a menudo expresados en percentiles.

Ambos el percentil y el cuartil son ejemplos de estadísticos de tipo rank, es decir, pueden ser calculados para datos con cualquier tipo de distribución. Son usados para resumir rápidamente la cantidad de datos ques e encuentran por debajo o por encima de un valor dado. Por ejemplo, el 50% de los datos se encuentran por debajo de la mediana y el otro 50% se encuentran por encima de la mediana.

# Cómo calcular los 5 números

Podemos calcular el valor de un percentil determinado haciendo uso de la función **percentile()** perteneciente a la librería Numpy. Podemos hacer uso de esta librería para obtener por ejemplo los valores de los  cuartiles. Esta función recibe como parámetros de entrada los datos y un array de los percentiles que queremos calcular.

In [9]:
from numpy import percentile
from numpy.random import seed
from numpy.random import randn

#Fijamos la semilla
seed(1)

#Generamos un conjunto de datos
data = 10*randn(10000) + 15

#Calculamos los cuartiles
quartiles = percentile(data, [25,50,75])

#Calculamos el máximo y el mínimo de nuestra muestra
val_min, val_max = data.min(), data.max()

#Vemos el resultado
print("El primer cuartil es: %.3f" % quartiles[0])
print("El segundo cuartil o mediana es: %.3f" % quartiles[1])
print("El tercer cuartil es: %.3f" % quartiles[2])
print("El mínimo es: %.3f" % val_min)
print("El máximo es: %.3f" % val_max)

El primer cuartil es: 8.371
El segundo cuartil o mediana es: 15.085
El tercer cuartil es: 21.718
El mínimo es: -21.564
El máximo es: 55.268


Lo que nos quiere decir este resultado es:

* El 25% de los valores de nuestra muestra de datos se encuentran por debjo del valor de 8.371 y por lo tanto el 75% de los valores de nuestra muestra se encuentran por encima de este valor. 

* El 50% de los valores de nuestra muestra se encuentran por debajo del valor 15.085 y por lo tanto el 50% de los valores de nuestra muestra se encuentran por encima de este valor.

* EL 75% de los valores de nuestra muestra se encuentran por debajo del valor 21.718 y por lo tanto el 25% de los valores de nuestra muestra se encuentran por encima de este valor.